overlord/configstate/configcore/kernel_test.go

```diff
@@ -0,0 +1,391 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/asserts/sysdb"
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+	"github.com/snapcore/snapd/bootloader/bootloadertest"
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/gadget"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord"
+	"github.com/snapcore/snapd/overlord/assertstate"
+	"github.com/snapcore/snapd/overlord/assertstate/assertstatetest"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/configstate"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/overlord/configstate/configcore"
+	"github.com/snapcore/snapd/overlord/devicestate"
+	"github.com/snapcore/snapd/overlord/devicestate/devicestatetest"
+	"github.com/snapcore/snapd/overlord/hookstate"
+	"github.com/snapcore/snapd/overlord/hookstate/hooktest"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/store/storetest"
+	"github.com/snapcore/snapd/sysconfig"
+	. "gopkg.in/check.v1"
+	"gopkg.in/tomb.v2"
+)
+
+type kernelSuite struct {
+	configcoreSuite
+	storetest.Store
+
+	overlord *overlord.Overlord
+
+	StoreSigning *assertstest.StoreStack
+	Brands       *assertstest.SigningAccounts
+}
+
+var _ = Suite(&kernelSuite{})
+
+var (
+	brandPrivKey, _ = assertstest.GenerateKey(752)
+)
+
+func (s *kernelSuite) SetUpTest(c *C) {
+	s.configcoreSuite.SetUpTest(c)
+
+	var err error
```

Comment:
(nitpick^9) is this needed, couldn't the next line just use: `err := ...`?

Reply:
Certainly not needed, but when I reuse a lot of times an err variable I prefer to declare, otherwise whenever I move/add checks it complains about missing declaration etc. and becomes annoying. Also look codes a bit more symmetric this way. But this is a personal preference, I can change if that does not fit the usual style in snapd.